src/reducers/transactionReducer.ts

```diff
@@ -53,21 +53,34 @@ export default function transactionReducer(state: TransactionState, action: Tran
     case TransactionActionTypes.RESET_ESTIMATED_FEE:
       return {
         ...state,
-        payloadError: null,
-        payload: null
+        estimatedFee: null,
+        estimatedFeeError: null
       };
     case TransactionActionTypes.SET_PAYLOAD:
       return {
         ...state,
         payloadError: action.payload.payloadError ? null : action.payload.payload,
         payload: action.payload.payload
       };
-    case TransactionActionTypes.CLEAR_PAYLOAD:
+    case TransactionActionTypes.RESET_PAYLOAD:
       return {
         ...state,
         payloadError: null,
         payload: null
       };
+    case TransactionActionTypes.RESET:
+      return {
+        ...state,
+        derivedReceiverAccount: null,
+        estimatedFee: null,
+        estimatedFeeError: null,
+        genericReceiverAccount: null,
+        receiverAddress: null,
+        unformattedReceiverAddress: null,
+        validationError: null,
```

Comment:
`validationError` in the context of `transactionReducer` suggests that this is some kind of transaction error. But afaict it is only related to receiver address.
IMHO this is really important, cause naming things properly makes the code much more future proof and easier to understand for future (both internal and external) developers.

Reply:
guilty ,  i'm not the best one naming variables :)